disktools.py

```diff
@@ -917,6 +924,22 @@ def utilityPartitions(self):
         # Return list of partition numbers for partitions we should preserve
         return [num for num in self.partitions.keys() if self.partitions[num]['id'] == self.ID_DELL_UTILITY]
 
+    def convertToGPT(self):
+        self.cmdWrap(['sgdisk', '--mbrtogpt', self.device])
+
+        gpt_tool = GPTPartitionTool(self.device)
+
+        # Copy hidden attribute on MS partitions
+        hidden_args = []
+        for num, part in self.partitions.iteritems():
+            if part['id'] in self.__HIDDEN_MS_IDS and gpt_tool.partitions[num]['id'] == gpt_tool.ID_LINUX:
+                hidden_args.append('--attributes=%d:set:62' % num)
+                gpt_tool.partitions[num]['hidden'] = True
+        if hidden_args:
+            self.cmdWrap([gpt_tool.SGDISK] + hidden_args + [self.device])
```

Comment:
Is this actually necessary? Won't the in-memory changes be written out later?

Reply:
Yes and no... depends. It's weird that the function will do half of the changes (the MBR conversion but not flags copying). It will force the caller to do a commit. Doing all changes is more consistent.

Reply:
Hmm, every partitioning change is a chance for something to go wrong so I would prefer to avoid doing unnecessary changes but I guess this is fine.